File: 2021/May/Day 26/sortByParity.py
# source : https://leetcode.com/problems/sort-array-by-parity-ii/submissions/
class Solution:
    def sortArrayByParityII(self, nums: List[int]) -> List[int]:


        odd = [i for i in nums if i%2==1];
        even = [i for i in nums if i%2==0];
        out = [];

        for i in range(len(odd)):
            out.append(even[i]);
            out.append(odd[i]);
        return out;

File: 2022/May/Day 10/sortByParity.cpp
// Source : https://leetcode.com/problems/sort-array-by-parity/
class Solution {
public:
    vector<int> sortArrayByParity(vector<int>& nums) {
        vector<int> temp;

        for(int i=0;i<size(nums);i++)
        {
            if(nums[i]%2==1){temp.insert(temp.end(),nums[i]);}
        }


          for(auto &i:temp)
          {nums.erase(std::remove(nums.begin(), nums.end(), i), nums.end());


          }


        for(auto&i:temp)
        {
            nums.insert(nums.end(),i);
        }

        return nums;





    }
};
